src/Agent.Worker/ContainerOperationProvider.cs

```diff
@@ -45,17 +43,19 @@ public async Task StartContainersAsync(IExecutionContext executionContext, objec
             // Check whether we are inside a container.
             // Our container feature requires to map working directory from host to the container.
             // If we are already inside a container, we will not able to find out the real working direcotry path on the host.
-#if OS_WINDOWS
-            // service CExecSvc is Container Execution Agent.
-            ServiceController[] scServices = ServiceController.GetServices();
-            if (scServices.Any(x => String.Equals(x.ServiceName, "cexecsvc", StringComparison.OrdinalIgnoreCase) && x.Status == ServiceControllerStatus.Running))
+            if (PlatformUtil.RunningOnWindows)
             {
-                throw new NotSupportedException(StringUtil.Loc("AgentAlreadyInsideContainer"));
+                // service CExecSvc is Container Execution Agent.
+                ServiceController[] scServices = ServiceController.GetServices();
+                if (scServices.Any(x => String.Equals(x.ServiceName, "cexecsvc", StringComparison.OrdinalIgnoreCase) && x.Status == ServiceControllerStatus.Running))
+                {
+                    throw new NotSupportedException(StringUtil.Loc("AgentAlreadyInsideContainer"));
+                }
             }
-#elif OS_RHEL6
+#if OS_RHEL6
             // Red Hat and CentOS 6 do not support the container feature
             throw new NotSupportedException(StringUtil.Loc("AgentDoesNotSupportContainerFeatureRhel6"));
```

Comment:
I am a bit surprised that the compiler doesn't error here when compiling on OS_RHEL6 since this throw makes the rest of the code dead code on that OS.

Reply:
I thought of this as well and was trying to rework it, but then saw that it would have been like that before.